### PR TITLE
firefox-sync-client: init at 1.8.0

### LINKED ISSUE #311378: Package request: firefox-sync-client
**Project description**

Adding this as a personal TODO to package it when I find the time.

A CLI for Firefox sync.

**Metadata**

* homepage URL: https://github.com/Mikescher/firefox-sync-client
* source URL: https://github.com/Mikescher/firefox-sync-client
* license: ASL2.0
* platforms: go platforms

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by ambroisie**
Closes #311378.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/fi/firefox-sync-client/package.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+buildGoModule rec {
+  pname = "firefox-sync-client";
+  version = "1.8.0";
+
+  src = fetchFromGitHub {
+    owner = "Mikescher";
+    repo = "firefox-sync-client";
+    rev = "v${version}";
+    sha256 = "sha256-Ax+v4a8bVuym1bp9dliXX85PXJk2Qlik3ME+adGiL1s=";
```
Comment (getchoo) [inline pkgs/by-name/fi/firefox-sync-client/package.nix:15]:
```suggestion
    rev = "refs/tags/v${version}";
    hash = "sha256-Ax+v4a8bVuym1bp9dliXX85PXJk2Qlik3ME+adGiL1s=";
```

File: pkgs/by-name/fi/firefox-sync-client/package.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+  vendorHash = "sha256-MYetPdnnvIBzrYrA+eM9z1P3+P5FumYKH+brvvlwkm4=";
+
+  meta = with lib; {
+    description = "Commandline-utility to list/view/edit/delete entries in a firefox-sync account.";
+    homepage = "https://github.com/Mikescher/firefox-sync-client";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ ambroisie ];
+    mainProgram = "ffsclient";
+  };
```
Comment (getchoo) [inline pkgs/by-name/fi/firefox-sync-client/package.nix:26]:
```suggestion
  meta = {
    description = "Commandline-utility to list/view/edit/delete entries in a firefox-sync account.";
    homepage = "https://github.com/Mikescher/firefox-sync-client";
    license = lib.licenses.asl20;
    maintainers = with lib.maintainers; [ ambroisie ];
    mainProgram = "ffsclient";
  };
```

**Review by getchoo — CHANGES_REQUESTED** [review]

File: pkgs/by-name/fi/firefox-sync-client/package.nix (hunk 1 above)
Comment (ambroisie) [reply pkgs/by-name/fi/firefox-sync-client/package.nix:15]: Ah! Should have thought about updating to `hash` before uploading to be fair.

I thought the idea was to keep using `rev = <tag>` and only use `refs/tags/<tag>` for upstreams that do clashing branch names.

File: pkgs/by-name/fi/firefox-sync-client/package.nix (hunk 2 above)
Comment (ambroisie) [reply pkgs/by-name/fi/firefox-sync-client/package.nix:26]: Sure, though IMHO `with lib` at `meta` level should be accepted as "neutral pattern".

File: pkgs/by-name/fi/firefox-sync-client/package.nix (hunk 1 above)
Comment (ambroisie) [reply pkgs/by-name/fi/firefox-sync-client/package.nix:15]: (`rev = <tag>` is what's used in the documentation, never `rev = refs/tags/<tag>`. So will keep it as-is).

Comment (getchoo) [reply pkgs/by-name/fi/firefox-sync-client/package.nix]: > (rev = <tag> is what's used in the documentation, never rev = refs/tags/<tag>. So will keep it as-is).

This doesn't mean it's correct. See https://github.com/NixOS/nixpkgs/issues/162454

It is also being [widely adopted](https://github.com/search?q=repo%3ANixOS%2Fnixpkgs%20%22refs%2Ftags%22&type=code) across the tree

**Review by fpletz — APPROVED** [review]

**Pull request merged**